### DISCUSSION TITLE
How to install jointjs+?

### DISCUSSION THREAD

**Discussion by makleo2002** [clientIO/joint#2636] [Q&A] [answered]
### Introduction

I downloaded the archive, moved the build folder from there to the project and wrote <script src="build/package/joint-plus.js"></script>
in index.html .
But I still can't access the Joint.ui
![изображение](https://github.com/clientIO/joint/assets/75272208/021f30b8-fc3a-402e-834c-d383f5644eb3)

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
As for `v4`, `joint-plus.js` already contains `@joint/core` (you should not load `@joint/core`).

Are there any errors in the console?

**Reply by makleo2002** [reply]
Was it necessary to register something in the console? Do I need to write something like npm install jointjs-plus? I downloaded the archive of the joint plus trial. Where should I drop the build folder?Why are there no detailed installation instructions.

**Reply by kumilingus** [reply, marked as answer]
> Was it necessary to register something in the console?

If a network error or JavaScript error occurs, it is displayed in the browser console automatically.

> Do I need to write something like npm install jointjs-plus? I downloaded the archive of the joint plus trial. Where should I drop the build folder?

First, get your copy of JointJS+:
1. If you have a valid license, download JointJS+ from [my.jointjs.com](https://my.jointjs.com). Alternatively, reach out to us for a [free 30-day trial](https://www.jointjs.com/free-trial).
2. Unzip the file and navigate to the `build/package` folder inside.
3. Copy the `joint-plus.tgz` file and paste it into the root of your package.

Then, install the JointJS+ package (`@joint/plus`) using your package manager:


```sh
npm add joint-plus.tgz
```

Now, you can import features of JointJS+ into your application as necessary:

```typescript
import { dia, shapes, ui } from '@joint/plus';
import '@joint/plus/build/joint-plus.css';
```

**Reply by makleo2002** [reply]
An error has appeared:
Uncaught SyntaxError: import declarations may only appear at top level of a module
To the index.js added type=module to index.html .
He writes that @joint/plus is a bare specifier, and when I insert a relative link, he writes Loading the module at "http://127.0.0.1:5500/diagramEditor/node_modules/@joint/plus " was blocked due to an unresolved MIME type ("text/html"). I tried to do without importing at all, and it kind of worked out. But I don't have Halo, selection and other things displayed. But at the same time, the CommandManager is working
![изображение](https://github.com/clientIO/joint/assets/75272208/2f17ea08-2a26-4378-a34c-64c51b6584a4)

**Reply by kumilingus** [reply]
Did you include `joint-plus.css`?

**Comment by ashoksrt-248** [comment]
what is the correct css import for joint js plus 4.1.1 version

**Reply by MartinKanera** [reply]
Hello, that would be `import '@joint/plus/joint-plus.css';`.

**Reply by ashoksrt-248** [reply]
earlier i used joint js 3.6.2 version then i am using imp;ort like 
@import '~@clientio/rappid/rappid.css';
@import '~bootstrap-icons/font/bootstrap-icons.css';

recently i updated jointjs plus 4.1.1 version so i changed to like this 
@import '~@joint/plus/joint-plus.css';
@import '~bootstrap-icons/font/bootstrap-icons.css';

After this change when i am refreshing or reloadinmg the page i am missing css for my buttons and dropdowns in my page.
please suggest me what i need to do here!...,Thanks in advance.

**Comment by ashoksrt-248** [comment]


**Reply by MartinKanera** [reply]
- By buttons and dropdowns, you are referring to the ui widgets provided by JointJS+?
- Are you setting a custom theme somewhere using the `setTheme()`?
- Can you verify that you can find the `joint-plus.css` file in the `node_modules` folder of your project?

**Reply by ashoksrt-248** [reply]
--> i found the joint-plus.css file in the node_modules folder of my project
 --> import { dia, shapes, ui, setTheme, linkTools, g, V, elementTools, util, highlighters } from '@joint/plus';( i am using setTheme from joint js only)
 --> we are mostly using bootstrap and jointjs css only

**Reply by MartinKanera** [reply]
What theme are you passing in as the argument to the `setTheme()`? Are all of the JointJS styles missing, or is there only an issue with some of the widgets?

**Reply by ashoksrt-248** [reply]
for buttons dropdown and tabs styles missing while reloading or refreshing or opening dataform through halo

**Reply by ashoksrt-248** [reply]
async startRappid() {
		setTheme('modern');
		this.initializePaper();
		this.initializeStencil();
		this.initializeSelection();
		this.initializeToolsAndInspector();
		this.initializeNavigator();
		this.initializeToolbar();
		this.initializeKeyboardShortcuts();
		this.initializeTooltips();
		this.setIsPaperRendered(true);
	}

**Reply by MartinKanera** [reply]
> for buttons dropdown and tabs styles missing while reloading or refreshing or opening dataform through halo

I don't understand. Does this mean that the styles are initially applied and then disappear after the page is reloaded?

Is there any error in the console that suggests importing from the wrong path? Can you inspect some JointJS UI plugin to see if the styles are applied to it? For example, using the modern theme in combination with `ui.Stencil` should yield these styles being applied to `.joint-stencil`: 

<img width="993" height="311" alt="image" src="https://github.com/user-attachments/assets/d82f8e88-98dc-40f6-b488-8e2f9bb666f3" /> 

You can tell where the styles are coming from by the CSS rule on the right with the origin file next to it (in this case `joint-plus.css`):

**Reply by ashoksrt-248** [reply]
Does this mean that the styles are initially applied and then disappear after the page is reloaded -- Yes

**Reply by kumilingus** [reply]
@ashoksrt-248 please share screenshots of the relevant part of the application (before/after).

There are no `tabs` or `dataforms` part of JointJS+.

**Reply by ashoksrt-248** [reply]
after the joint js upgradation only this issue is coming

**Reply by kumilingus** [reply]
Ok. Please share the screenshots.
